### PR TITLE
Reorganize WIT inference utility code

### PR DESCRIPTION
**Pull request by jameswex**
* Motivation for features / changes

Some python code reorganization to maximize code reuse in upcoming WIT jupyter/colab notebook mode code.

* Technical description of changes

Moved some logic from interactive_inference_plugin.py to inference_utils.py

Refactored PredictResponse conversion code so that much of it can be reused in upcoming notebook mode to convert from raw tensor arrays to ClassificationResponse and RegressionResponse.

Removed unnecessary platform_utils dependence on tensorboard.util (as platform_utils already depended on tensorflow which could do the same work)

* Screenshots of UI changes

No UI changes

* Detailed steps to verify changes work correctly (as executed by you)

Run 4 WIT demos and verify inference still works

Run WIT in Tensorboard with classification and regression models and verify inference still works

* Alternate designs / implementations considered

N/A

### CODE REVIEW

**Comment by jameswex** [comment]
@tolga-b please review

File: tensorboard/plugins/interactive_inference/utils/platform_utils.py [hunk 1]
```diff
@@ -188,7 +187,7 @@ def call_servo(examples, serving_bundle):
 
   if serving_bundle.use_predict:
     request.inputs[serving_bundle.predict_input_tensor].CopyFrom(
-        tensor_util.make_tensor_proto(
+        tf.compat.v1.make_tensor_proto(
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply tensorboard/plugins/interactive_inference/utils/platform_utils.py]: Also, what can we do to move off of compat.v1? The compat.v1 won't go away but it generally is a good idea to move off of old API.

File: tensorboard/plugins/interactive_inference/utils/inference_utils.py [hunk 2]
```diff
@@ -597,3 +601,151 @@ def get_example_features(example):
...
+    A ClassificationResponse or RegressionResponse proto.
+  """
+  batch_size = 64
+  if serving_bundle.estimator and serving_bundle.feature_spec:
+    # If provided an estimator and feature spec then run inference locally.
+    preds = serving_bundle.estimator.predict(
+      lambda: tf.data.Dataset.from_tensor_slices(
+        tf.parse_example([ex.SerializeToString() for ex in examples],
+        serving_bundle.feature_spec)).batch(batch_size))
+    preds_key = 'probabilities'
```
[the remark this answers is not in the document]
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/utils/inference_utils.py]: done

File: tensorboard/plugins/interactive_inference/utils/platform_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jameswex) [reply tensorboard/plugins/interactive_inference/utils/platform_utils.py]: added comment. we could write our own logic for converting from tf.example to tensor but the existing method in TF looks a bit complex.

File: tensorboard/plugins/interactive_inference/utils/common_utils.py [hunk 3]
```diff
@@ -49,22 +49,43 @@ def convert_predict_response(pred, serving_bundle):
   """
   output = pred.outputs[serving_bundle.predict_output_tensor]
   raw_output = output.float_val
-  output_index = 0
+  if serving_bundle.model_type == 'classification':
```
Comment (stephanwlee) [inline tensorboard/plugins/interactive_inference/utils/common_utils.py]: Would you prefer using enum for model_type instead of hard coding strings?

**Pull request merged**